traits/ctraits.c

```diff
@@ -4223,34 +4229,60 @@ _trait_delegate(trait_object *trait, PyObject *args)
 |  Sets the appropriate value comparison mode flags of a CTrait instance:
 +----------------------------------------------------------------------------*/
 
-static PyObject *
-_trait_comparison_mode(trait_object *trait, PyObject *args)
+static int
+_set_trait_comparison_mode(trait_object *trait, PyObject *value, void *closure)
 {
-    int comparison_mode;
-
-    if (!PyArg_ParseTuple(args, "i", &comparison_mode)) {
-        return NULL;
+    long comparison_mode = PyLong_AsLong(value);
+    
+    if (comparison_mode == -1 && PyErr_Occurred() != NULL) {
```

Comment:
Slightly more idiomatic:

```suggestion
    if (comparison_mode == -1 && PyErr_Occurred()) {
```